packages/zend-barcode/library/Zend/Barcode/Renderer/Image.php

```diff
@@ -151,26 +151,25 @@ public function getWidth()
      * Set an image resource to draw the barcode inside
      *
      * @param $image
-     * @return Zend_Barcode_Renderer
+     * @return self
      * @throws Zend_Barcode_Renderer_Exception
      */
     public function setResource($image)
     {
-        if (gettype($image) != 'resource' || get_resource_type($image) != 'gd') {
-            // require_once 'Zend/Barcode/Renderer/Exception.php';
-            throw new Zend_Barcode_Renderer_Exception(
-                'Invalid image resource provided to setResource()'
-            );
+        if ((gettype($image) == 'resource' && get_resource_type($image) == 'gd') || get_class($image) === 'GdImage') {
```

Comment:
```suggestion
        if ((gettype($image) === 'resource' && get_resource_type($image) === 'gd') || get_class($image) === 'GdImage') {
```

if the line is already changed, upgrade code style to use strict compare